smartmoneyconcepts/SMC.py

```diff
@@ -121,34 +122,83 @@ def highs_lows(cls, ohlc: DataFrame) -> Series:
                     previous_high = ohlc["high"][i - 1]
                     current_high = ohlc["high"][i]
                     next_high = ohlc["high"][i + 1]
-                    if (previous_high > current_high and highs_lows[i-1] == 0) or (next_high > current_high and highs_lows[i+1] == 0):
+                    if (previous_high > current_high and highs_lows[i - 1] == 0) or (
+                        next_high > current_high and highs_lows[i + 1] == 0
+                    ):
                         highs_lows[i] = 0
                         still_adjusting = True
-                        if (previous_high > next_high and highs_lows[i-1] == 0):
+                        if previous_high > next_high and highs_lows[i - 1] == 0:
                             highs_lows[i - 1] = 1
                         else:
                             highs_lows[i + 1] = 1
                 if highs_lows[i] == -1:
                     previous_low = ohlc["low"][i - 1]
                     current_low = ohlc["low"][i]
                     next_low = ohlc["low"][i + 1]
-                    if (previous_low < current_low and highs_lows[i-1] == 0) or (next_low < current_low and highs_lows[i+1] == 0):
+                    if (previous_low < current_low and highs_lows[i - 1] == 0) or (
+                        next_low < current_low and highs_lows[i + 1] == 0
+                    ):
                         highs_lows[i] = 0
                         still_adjusting = True
-                        if (previous_low < next_low and highs_lows[i-1] == 0):
+                        if previous_low < next_low and highs_lows[i - 1] == 0:
                             highs_lows[i - 1] = -1
                         else:
                             highs_lows[i + 1] = -1
 
-        levels = np.where(highs_lows != 0, np.where(highs_lows == 1, ohlc["high"], ohlc["low"]), np.nan)
+        levels = np.where(
+            highs_lows != 0,
+            np.where(highs_lows == 1, ohlc["high"], ohlc["low"]),
+            np.nan,
+        )
 
         highs_lows = pd.Series(highs_lows, name="HighsLows")
         levels = pd.Series(levels, name="Levels")
 
         return pd.concat([highs_lows, levels], axis=1)
 
     @classmethod
-    def bos_choch(cls, ohlc: DataFrame, close_break=True, filter_liquidity=False) -> Series:
+    def swing_tops_bottoms(
```

Comment:
Can you create a test for this function?

Reply:
done